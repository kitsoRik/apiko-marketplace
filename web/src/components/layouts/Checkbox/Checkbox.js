import React from 'react';
import Label from '../Label';


import './Checkbox.scss';

const Checkbox = ({ value, checked = false, className, ...props }) => {
    return (
        <div className={`checkbox ${className}`}>
            <input checked={checked} className="checkbox-input" type="checkbox" {...props} />
            <Label className="checkbox-label" value={value} />
        </div>
    );
}

export default Checkbox;