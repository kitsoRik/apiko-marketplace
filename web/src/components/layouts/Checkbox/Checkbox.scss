.checkbox {
    display: inline-flex;
    align-items: center;    

    &-input {
        margin-right: 10px;
        
        width: 24px;
        height: 24px;

        position: relative;
        
        display: flex;
        justify-content: center;

        border: 1px solid gray;
        border-radius: 4px;

        cursor: pointer;

        transition: all 175ms cubic-bezier(0.1, 0.1, 0.25, 1);
        
        appearance: none;

        &::before {
            content: '';

            width: 8px;
            height: 14px;

            position: absolute;

            display: block;

            border-style: solid;
            border-color: white;
            border-width: 0 2px 2px 0;

            transform: rotate(45deg);

            opacity: 0;
          }
          
          &:checked {
            color: white;
            background: #349A89;
            border-color: #349A89;

            &::before {
              opacity: 1;
            }
          }
        }

    &-label {

    }
}