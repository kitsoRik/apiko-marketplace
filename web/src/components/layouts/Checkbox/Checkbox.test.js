import React from "react";
import { create } from "react-test-renderer";
import Checkbox from "./Checkbox";

describe("Checkbox", () => {
	it("should render without value", () => {
		const component = create(<Checkbox />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with value", () => {
		const component = create(<Checkbox value="My custom value" />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with value and checked option", () => {
		const component = create(
			<Checkbox value="My custom value" checked={false} />
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
});
