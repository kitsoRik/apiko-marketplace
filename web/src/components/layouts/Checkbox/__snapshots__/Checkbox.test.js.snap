// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render with value 1`] = `
<div
  className="checkbox undefined"
>
  <input
    checked={false}
    className="checkbox-input"
    type="checkbox"
  />
  <span
    className="label-solo-value checkbox-label"
    error={null}
  >
    My custom value
  </span>
</div>
`;

exports[`Checkbox should render with value and checked option 1`] = `
<div
  className="checkbox undefined"
>
  <input
    checked={false}
    className="checkbox-input"
    type="checkbox"
  />
  <span
    className="label-solo-value checkbox-label"
    error={null}
  >
    My custom value
  </span>
</div>
`;

exports[`Checkbox should render without value 1`] = `
<div
  className="checkbox undefined"
>
  <input
    checked={false}
    className="checkbox-input"
    type="checkbox"
  />
  <span
    className="label-solo-value checkbox-label"
    error={null}
  >
    
  </span>
</div>
`;
